coalib/bearlib/aspects/Formatting.py

```diff
@@ -0,0 +1,395 @@
+        declaration.
+        """
+    blank_lines_after_declarations = Taste[int](
+        'Represents the number of blank lines after declarations',
+        (0, 1, 2), default=0)
+
+
+@BlankLine.subaspect
+class BlankLineAfterProcedure:
+    """
+    Blank lines after procedures` refer to the blank lines found after
+    procedures or functions.
+    """
+    class docs:
+        example = """
+        void proc(void){
+            printf("this does nothing");
+        }
+
+        int add(float a, float b){
+            return a + b;
+        }
+        """
+        example_language = 'C++'
+        importance_reason = """
+        Having a specific and reasonable number of blank lines after every
+        procedures improves on the readability of the code.
+        """
+        fix_suggestions = """
+        `BlankLintAfterProcedure` issues can be fixed  specifying (and of
+        course using) a reasonable number of blank lines to use after
+        procedures' definition.
+        """
+    blank_lines_after_procedures = Taste[int](
+        'Represents the number of blank lines to use after a procedure or a'
+        ' function', (0, 1, 2), default=1)
+
+
+@BlankLine.subaspect
+class BlankLineAfterClass:
+    """
+    Blank lines after classes refer to the blank lines found after classes'
+    definitions.
+    """
+    class docs:
+        example = """
+        class SomeClass:
+            def __init__(self):
+                raise RuntimeError('Never instantiate this class')
+
+
+        def func():
+            pass
+        """
+        example_language = 'python'
+        importance_reason = """
+        Having a specific number of blank lines after every classes'
+        definitions declarations improves on the readability of the code.
+        """
+        fix_suggestions = """
+        """
+    blank_lines_after_class = Taste[int](
+        'Represents the number of blank lines to use after a class'
+        'definition.', (0, 1, 2, 3), default=2)
+
+
+@BlankLine.subaspect
+class NewlineAtEOF:
+    """
+    A newline character (usually '\n', aka CR or CRLF) at  found at the end of
+    file.
+    """
+    class docs:
+        example = """
+        def do_nothing():
+            pass\n
+        """
+        example_language = 'python'
+        importance_reason = """
+        A text file, under unix, consists of a series of lines, each of which
+        ends with a newline character (\\n). A file that is not empty and does
+        not end with a newline is therefore not a text file.
+
+        It's not just bad style, it can lead to unexpected behavior, utilities
+        that are supposed to operate on text files may not cope well with files
+        that don't end with a newline.
+        """
+        fix_suggestions = """
+        `NewlineAtEOF` issues can be fixed by simply adding a newline at the
+        end of the file.
+        """
+    enforce_newline_at_EOF = Taste[bool](
```

Comment:
never start a taste with "enforce" an aspect always enforces something, the taste helps defining what exactly; it never adds an aspect to the enforcement